backend/fms_core/services/container.py

```diff
@@ -139,7 +139,12 @@ def get_or_create_container(barcode, kind=None, name=None, coordinates=None, con
                 errors.append(f"Container kind is required to create a container.")
             elif container_parent and CONTAINER_KIND_SPECS[container_parent.kind].requires_coordinates and not coordinates:
                 errors.append(f"Parent container kind {container_parent.kind} requires that you provide coordinates.")
+            elif name is None and Container.objects.filter(name=barcode):
+                    errors.append(f"Missing container name, barcode will replace container name. Container with name {barcode} already exists")
```

Comment:
Start with the error part. Finish your error message with a point.

Since you are using the Service for the form creation you may want some bad entry hit the DB and generate a validation error for the form to catch. The validation error for fields with the same name as fields in the form are mapped to it and displayed. You can also patch it up in the viewset by creating your own validation errors. It might be nice in this case to actually have some more to go on that name already exists with an empty field above.

Its important to make sure that all case are caught and the operation is rolled back in case of bad entry.